app/questionnaire/value_source_resolver.py

```diff
@@ -23,7 +23,7 @@ class ValueSourceResolver:
     list_store: ListStore
     metadata: dict
     schema: QuestionnaireSchema
-    location: Union[Location, RelationshipLocation]
+    location: Union[Optional[Location], RelationshipLocation]
```

Comment:
When we have more than one type, we should use `None` instead of `Optional` as per our type hinting style guide.

Reply:
If `location` can be `None` then the resolver code needs to be tweaked to handle location not existing, it currently doesn't and would fail if it didn't have a location when trying to access location's attribute.

Reply:
This is a slightly odd situation. I am thinking that post_submission would be where location is none, but at that point get_schema_value would not be called from a field_handler??